cdc/owner/ddl_manager.go

```diff
@@ -598,6 +624,53 @@ func (m *ddlManager) cleanCache() {
 	m.physicalTablesCache = nil
 }
 
+func (m *ddlManager) checkAndBootstrap(ctx context.Context) (bool, error) {
+	if atomic.LoadInt32(&m.bootstrapState) == int32(bootstrapStateCompleted) {
+		return true, nil
+	}
+
+	select {
+	case err := <-m.errCh:
+		return false, err
+	default:
+	}
+
+	if atomic.LoadInt32(&m.bootstrapState) == int32(bootstrapStateRunning) {
+		return false, nil
+	}
+	// begin bootstrap
+	atomic.StoreInt32(&m.bootstrapState, int32(bootstrapStateRunning))
+	tables, err := m.allTables(ctx)
+	if err != nil {
+		return false, err
+	}
+	var bootstrapEvents []*model.DDLEvent
+	for _, table := range tables {
+		ddlEvent := &model.DDLEvent{
+			StartTs:     m.startTs,
+			CommitTs:    m.startTs,
+			TableInfo:   table,
+			IsBootstrap: true,
+		}
+		bootstrapEvents = append(bootstrapEvents, ddlEvent)
+	}
+	// send bootstrap events
+	go func() {
```

Comment:
Why do we have to send bootstrap events in another goroutine?

Reply:
Otherwise it will block the whole `Owner.Tick()` loop.

Reply:
It looks cannot guarantee that all bootstrap events are sent before other event ?

Reply:
No, it can.
If `!bootstrapStateCompleted`, the ddlBarrierTs would not advance by ddlManager, so other event will not be sent.

Reply:
> Otherwise it will block the whole `Owner.Tick()` loop.

Will the send DDL event block the owner?

IMHO, sending Bootstrap event is just like sending DDL event.